File: frontend/types/transcript.ts
// Purpose: Shared transcript types for Vox frontend.
// Callers: API helpers, editors, media components.
// Deps: None.
// API: Segment, Summary, TranscriptionResult.
// Side effects: None.
export type Segment = { id: number; start: number; end: number; text: string };
export type Summary = { short: string; key_points: string[]; action_items: string[] };
export type Downloads = Record<"txt" | "srt" | "vtt" | "json" | "csv", string>;
export type TranscriptionResult = { file_id: string; language: string; duration: number; segments: Segment[]; transcript: string; summary: Summary; downloads: Downloads };
export type UploadResponse = { file_id: string; filename: string; file_type: "audio" | "video"; status: "uploaded" };

File: backend/app/schemas/transcript.py
"""Purpose: Transcript result schemas.
Callers: Transcription and segment update routes.
Deps: Pydantic, Segment schema.
API: Summary, TranscriptionResponse, UpdateSegmentsRequest.
Side effects: None.
"""
from pydantic import BaseModel

from app.schemas.segment import Segment


class Summary(BaseModel):
    short: str
    key_points: list[str]
    action_items: list[str]


class TranscriptionResponse(BaseModel):
    file_id: str
    language: str
    duration: float
    segments: list[Segment]
    transcript: str
    summary: Summary
    downloads: dict[str, str]


class UpdateSegmentsRequest(BaseModel):
    file_id: str
    segments: list[Segment]
